### ISSUE TITLE
[Bug] Unclear problem description claiming non-existent conflict between JNSQ and Scatterer

### ISSUE THREAD

**Issue by TheDancingFerret** [KSP-CKAN/CKAN#3789]
Background
----------
* Operating System: Windows 10
* CKAN Version: 1.31.2
* KSP Version: 1.12.5

Problem
-------
I have manually installed JNSQ and Scatterer 3:v0.0825b and modified them to be compatible with each other. However, when I attempt to use CKAN to install another, unrelated mod (in this case Planetside Exploration Technologies, although this will occur with any mod) CKAN will throw an error saying that JNSQ and Scatterer 3:v0.0825b+ are incompatible and refuse to proceed.

**Comment by HebaruSan** [comment]
A manually installed JNSQ cannot be detected because there is no `JNSQ.dll` file. So if CKAN is reporting a conflict with JNSQ, then JNSQ was installed with CKAN, not manually.

And JNSQ does not have any conflicts with Scatterer, so something else is going on here that cannot be inferred from the description.

A screenshot of the problem would be helpful.

**Comment by TheDancingFerret** [comment]
![ckan_Wc1Tn6riWn](https://user-images.githubusercontent.com/42417220/220804631-d982ab6e-ec38-44cf-a9b9-1f27c49c6518.png)

Attached the specific message CKAN gives.

Thinking about it more, it is possible JNSQ was installed via CKAN. I have GEP, GPP, and most of their related mods installed and I did have to manually reinstall and tweak them to get them working.

Scatterer version .08 and up turn most atmospheric planets in JNSQ/GPP/GEP into black holes. Team Galileo recommends using an older version of Scatterer which does not cause this issue.

**Comment by TheDancingFerret** [comment]
Well maybe suspend this for a bit. Been fighting this issue for days and only just now realized consciously that it only mentions GEP and GPP.

/facepalm

**Comment by HebaruSan** [comment]
Yes, those two mods _do_ conflict with that version of Scatterer, because they haven't been updated to support it.

**Comment by TheDancingFerret** [comment]
JNSQ's metadata should be updated to show this compatibility as well, as it has the same issue as GEP and GPP.

The fundamental issue I'm facing is the same though. CKAN is not installing a totally unrelated mod because of this.

**Comment by HebaruSan** [comment]
JNSQ's metadata is maintained by its authors, you can request changes here (but they're not very active):

- <https://github.com/Galileo88/JNSQ/blob/master/CKAN/JNSQ.netkan>

CKAN is _supposed to_ abort when there are conflicts between already-installed mods.

**Comment by TheDancingFerret** [comment]
So it is intended behavior for CKAN to refuse to install a mod because it thinks that two other unrelated mods that it isn't managing are incompatible?

**Comment by HebaruSan** [comment]
"that it isn't managing" is incorrect. CKAN is managing GEP and GPP and Scatterer in this install.

Yes, an inconsistent install needs to be fixed before we mess with it even more.

**Comment by TheDancingFerret** [comment]
All three mods were manually installed.

**Comment by HebaruSan** [comment]
If that was true, none of those messages would be getting printed. Conflicts are not enforced for manually installed mods. Maybe you're remembering some other time when they actually were manually installed, but that's not this time.

**Comment by HebaruSan** [comment]
Also, none of these mods can be auto-detected when manually installed, because they don't have plugins. 100% confirmation of CKAN-installed.

**Comment by TheDancingFerret** [comment]
![ckan_DftgNWGaJS](https://user-images.githubusercontent.com/42417220/220810535-98742092-014e-480a-9f8f-7737d75d2bdd.png)
![ckan_jjkxorCWox](https://user-images.githubusercontent.com/42417220/220810547-72141456-1267-42de-a532-9294d07c7176.png)

CKAN definitely can recognize mods it didn't install. 

CKAN shows the Galileo suite of mods as unchecked, but it still throws the error I linked above when I try to install anything.

Just reopen this and let other people look at it. Obviously there is something odd here. It may be an issue on my end, or it CKAN recognizing manually installed mods when it shouldn't.

I initially uninstalled everything in the Galileo suite and reinstalled them manually in hopes of sidestepping this exact issue. It didn't work. Based on what you are saying about how CKAN handles manually installed mods, then CKAN is either detecting them when it isn't supposed to, or when I uninstalled them it left metadata of some kind that caused it to continue managing them after I manually reinstalled. Either way suggests a bug based on what you are saying it should be doing.

**Comment by HebaruSan** [comment]
OK, _Scatterer_ does have a plugin, so it _is_ detected when manually installed.

When you "uninstalled everything in the Galileo suite", how did you do that? Did you uncheck the box and let CKAN uninstall, or did you just start deleting files?

**Comment by TheDancingFerret** [comment]
I uninstalled them using CKAN. There were a few files left after CKAN did its thing, so they were deleted manually.

**Comment by HebaruSan** [comment]
Let's see your `<KSP>/CKAN/registry.json` file. That should confirm what CKAN thinks is and is not installed.

**Comment by TheDancingFerret** [comment]
https://drive.google.com/file/d/187N8NkkGIrOerykS2HmqqJgy0YhJ20hA/view?usp=sharing

Sorry, its larger than GitHub allows.

**Comment by HebaruSan** [comment]
OK, this is the relevant part:

```json
"installed_dlls": {
"999-Scale-Redist": "GameData/999_Scale_Redist.dll",
"ModuleManager": "GameData/ModuleManager.4.2.2.dll",
"000-AT-Utils": "GameData/000_AT_Utils/Plugins/000_AT_Utils.dll",
"001-AnisotropicPartResizer": "GameData/000_AT_Utils/Plugins/001_AnisotropicPartResizer.dll",
"002-MultiAnimators": "GameData/000_AT_Utils/Plugins/002_MultiAnimators.dll",
"0-00-AT-Utils-UI": "GameData/000_AT_Utils/Plugins/0_00_AT_Utils_UI.dll",
"CC": "GameData/000_AT_Utils/Plugins/CC.UI.dll",
"ConfigurableContainers": "GameData/000_AT_Utils/Plugins/ConfigurableContainers.dll",
"SubmodelResizer": "GameData/000_AT_Utils/Plugins/SubmodelResizer.dll",
"ClickThroughBlocker": "GameData/000_ClickThroughBlocker/Plugins/ClickThroughBlocker.dll",
"FilterExtensions": "GameData/000_FilterExtensions/Plugins/FilterExtensions.dll",
"0Harmony": "GameData/000_Harmony/0Harmony.dll",
"HarmonyInstallChecker": "GameData/000_Harmony/HarmonyInstallChecker.dll",
"KSPBurst": "GameData/000_KSPBurst/Plugins/KSPBurst.dll",
"Microsoft": "GameData/000_KSPBurst/Plugins/Microsoft.Extensions.FileSystemGlobbing.dll",
"System": "GameData/MechJeb2/Plugins/System.Buffers.dll",
"Unity": "GameData/000_KSPBurst/Plugins/Unity.Mathematics.dll",
"BlendshapeModelLoader": "GameData/000_TexturesUnlimited/Plugins/BlendshapeModelLoader.dll",
"TexturesUnlimited": "GameData/000_TexturesUnlimited/Plugins/TexturesUnlimited.dll",
"aaa-Toolbar": "GameData/000_Toolbar/Plugins/aaa_Toolbar.dll",
"ToolbarControl": "GameData/001_ToolbarControl/Plugins/ToolbarControl.dll",
"Attached": "GameData/999_KSP-Recall/Plugins/Attached.dll",
"AttachedOnEditor": "GameData/999_KSP-Recall/Plugins/AttachedOnEditor.dll",
"ChillingOut": "GameData/999_KSP-Recall/Plugins/ChillingOut.dll",
"Driftless": "GameData/999_KSP-Recall/Plugins/Driftless.dll",
"KSP-Recall": "GameData/999_KSP-Recall/Plugins/KSP-Recall.dll",
"KSPe": "GameData/TweakScale/Plugins/KSPe.Light.TweakScale.dll",
"Refunding": "GameData/999_KSP-Recall/Plugins/Refunding.dll",
"Resourceful": "GameData/999_KSP-Recall/Plugins/Resourceful.dll",
"AntennaHelper": "GameData/AntennaHelper/Plugins/AntennaHelper.dll",
"ASETPropsIdentifier": "GameData/ASET/ASET_Props/Plugin/ASETPropsIdentifier.dll",
"Astrogator": "GameData/Astrogator/Plugins/Astrogator.dll",
"B9PartSwitch": "GameData/B9PartSwitch/Plugins/B9PartSwitch.dll",
"BetterTimeWarpCont": "GameData/BetterTimeWarp/Plugins/BetterTimeWarpCont.dll",
"Chatterer": "GameData/Chatterer/Plugins/Chatterer.dll",
"CCK": "GameData/CommunityCategoryKit/CCK.dll",
"CommunityTraitIcons": "GameData/CommunityTraitIcons/CommunityTraitIcons.dll",
"ConformalDecals": "GameData/ConformalDecals/Plugins/ConformalDecals.dll",
"CC-RemoteTech": "GameData/ContractConfigurator/CC_RemoteTech.dll",
"ContractConfigurator": "GameData/ContractConfigurator/ContractConfigurator.dll",
"KerKonConConExt": "GameData/ContractConfigurator/KerKonConConExt.dll",
"CrewManifest": "GameData/CrewManifest/Plugins/CrewManifest.dll",
"SimpleBoiloff": "GameData/CryoTanks/Plugins/SimpleBoiloff.dll",
"DeployableEngines": "GameData/DeployableEngines/Plugins/DeployableEngines.dll",
"DistantObject": "GameData/DistantObject/Plugins/DistantObject.dll",
"MeshEngine": "GameData/DistantObject/Plugins/PluginData/MeshEngine.dll",
"MeshEngineStock": "GameData/DistantObject/Plugins/PluginData/MeshEngineStock.dll",
"MeshEngineTweakScale": "GameData/DistantObject/Plugins/PluginData/MeshEngineTweakScale.dll",
"DMagic": "GameData/DMagicOrbitalScience/Plugins/DMagic.dll",
"DynamicBatteryStorage": "GameData/DynamicBatteryStorage/Plugins/DynamicBatteryStorage.dll",
"EditorExtensionsRedux": "GameData/EditorExtensionsRedux/Plugins/EditorExtensionsRedux.dll",
"Atmosphere": "GameData/EnvironmentalVisualEnhancements/Plugins/Atmosphere.dll",
"CelestialShadows": "GameData/EnvironmentalVisualEnhancements/Plugins/CelestialShadows.dll",
"CityLights": "GameData/EnvironmentalVisualEnhancements/Plugins/CityLights.dll",
"EVEManager": "GameData/EnvironmentalVisualEnhancements/Plugins/EVEManager.dll",
"PartFX": "GameData/EnvironmentalVisualEnhancements/Plugins/PartFX.dll",
"PQSManager": "GameData/EnvironmentalVisualEnhancements/Plugins/PQSManager.dll",
"ShaderLoader": "GameData/EnvironmentalVisualEnhancements/Plugins/ShaderLoader.dll",
"Terrain": "GameData/EnvironmentalVisualEnhancements/Plugins/Terrain.dll",
"TextureConfig": "GameData/EnvironmentalVisualEnhancements/Plugins/TextureConfig.dll",
"Utils": "GameData/EnvironmentalVisualEnhancements/Plugins/Utils.dll",
"BuildManager": "GameData/EnvironmentalVisualEnhancements/Plugins/_BuildManager.dll",
"FarFutureTechnologies": "GameData/FarFutureTechnologies/Plugins/FarFutureTechnologies.dll",
"Firespitter": "GameData/Firespitter/Plugins/Firespitter.dll",
"FMRSContinued": "GameData/FMRS/Plugins/FMRSContinued.dll",
"ForScience": "GameData/ForScience/Plugins/ForScience.dll",
"FreeIva": "GameData/FreeIva/Plugins/FreeIva.dll",
"HangerExtenderExtended": "GameData/FShangarExtender/Plugins/HangerExtenderExtended.dll",
"MiniAVC": "GameData/RESCALE/MiniAVC.dll",
"SigmaHeatShifter": "GameData/GEP/GEP_Plugins/HeatShifter/Plugins/SigmaHeatShifter.dll",
"SigmaTweakChutes": "GameData/JNSQ/JNSQ_Plugins/Sigma/TweakChutes/Plugins/SigmaTweakChutes.dll",
"GPPTextureChecker": "GameData/GPP/GPP_InstallationChecker/GPPTextureChecker.dll",
"KerbalRenamer": "GameData/GPP/GPP_Renamer/KerbalRenamer.dll",
"SuitProg": "GameData/GPP/GPP_SuitProgression/SuitProg.dll",
"Sigma88LoadingScreens": "GameData/JNSQ/LoadingScreens/Sigma88LoadingScreens.dll",
"Hangar": "GameData/Hangar/Plugins/Hangar.dll",
"HLA": "GameData/HLAirshipsCore/Plugins/HLA.WatchDog.dll",
"HLAirships": "GameData/HLAirshipsCore/Plugins/HLAirships.dll",
"HullcamVDSContinued": "GameData/HullCameraVDS/Plugins/HullcamVDSContinued.dll",
"JanitorsCloset": "GameData/JanitorsCloset/Plugins/JanitorsCloset.dll",
"JSIPartUtilities": "GameData/JSI/PartUtilities/Plugins/JSIPartUtilities.dll",
"RasterPropMonitor": "GameData/JSI/RasterPropMonitor/Plugins/RasterPropMonitor.dll",
"KAS-API-v2": "GameData/KAS/Plugins/KAS-API-v2.dll",
"KAS": "GameData/KAS/Plugins/KAS.dll",
"KSPDev-Utils": "GameData/KAS/Plugins/KSPDev_Utils.2.7.dll",
"KerbalEngineer": "GameData/KerbalEngineer/KerbalEngineer.Unity.dll",
"KerbalJointReinforcement": "GameData/KerbalJointReinforcement/Plugin/KerbalJointReinforcement.dll",
"DeployableAeroSurfaces": "GameData/KerbalReusabilityExpansion/DeployableAeroSurfaces.dll",
"KerbalStats": "GameData/KerbalStats/Plugins/KerbalStats.dll",
"HyperEdit": "GameData/Kerbaltek/HyperEdit.dll",
"Kopernicus": "GameData/Kopernicus/Plugins/Kopernicus.Parser.dll",
"Kronometer": "GameData/Kronometer/Plugins/Kronometer.dll",
"0-MultipleModuleInPartAPI": "GameData/KSPCommunityFixes/Plugins/0_MultipleModuleInPartAPI.dll",
"KSPCommunityFixes": "GameData/KSPCommunityFixes/Plugins/KSPCommunityFixes.dll",
"KSPWheel": "GameData/KSPWheel/Plugin/KSPWheel.dll",
"MechJeb2": "GameData/MechJeb2/Plugins/MechJeb2.dll",
"AvionicsSystems": "GameData/MOARdV/AvionicsSystems/AvionicsSystems.dll",
"MoonSharp": "GameData/MOARdV/AvionicsSystems/MoonSharp.Interpreter.dll",
"ModularFlightIntegrator": "GameData/ModularFlightIntegrator/ModularFlightIntegrator.dll",
"DockingPortAlignmentIndicator": "GameData/NavyFish/Plugins/Docking Port Alignment Indicator/DockingPortAlignmentIndicator.dll",
"DPAI-RPM": "GameData/NavyFish/Plugins/Docking Port Alignment Indicator/DPAI_RPM.dll",
"ModuleDockingNodeNamed": "GameData/NavyFish/Plugins/Docking Port Alignment Indicator/ModuleDockingNodeNamed.dll",
"NearFutureElectrical": "GameData/NearFutureElectrical/Plugins/NearFutureElectrical.dll",
"NearFutureExploration": "GameData/NearFutureExploration/Plugins/NearFutureExploration.dll",
"NFPropUtils": "GameData/NearFutureProps/Plugins/NFPropUtils.dll",
"NearFuturePropulsion": "GameData/NearFuturePropulsion/Plugins/NearFuturePropulsion.dll",
"NearFutureSolar": "GameData/NearFutureSolar/Plugins/NearFutureSolar.dll",
"NearFutureUtils": "GameData/NearFutureSpacecraft/Plugins/NearFutureUtils.dll",
"FinalFrontier": "GameData/Nereid/FinalFrontier/Plugins/FinalFrontier.dll",
"Grass": "GameData/Parallax/Grass.dll",
"ParallaxOptimized": "GameData/Parallax/ParallaxOptimized.dll",
"ParallaxQualityLibrary": "GameData/Parallax/ParallaxQualityLibrary.dll",
"PlanetarySurfaceStructures": "GameData/PlanetaryBaseInc/BaseSystem/Plugins/PlanetarySurfaceStructures.dll",
"PlanetShine": "GameData/PlanetShine/Plugins/PlanetShine.dll",
"PortraitStats": "GameData/PortraitStats/PortraitStats.dll",
"RecoveryController": "GameData/RecoveryController/Plugins/RecoveryController.dll",
"Restock": "GameData/ReStock/Plugins/Restock.dll",
"Reviva": "GameData/Reviva/Reviva.dll",
"RocketSoundEnhancement": "GameData/RocketSoundEnhancement/Plugins/RocketSoundEnhancement.Unity.dll",
"SCANmechjeb": "GameData/SCANsat/Plugins/SCANmechjeb.dll",
"SCANsat": "GameData/SCANsat/Plugins/SCANsat.Unity.dll",
"Scatterer": "GameData/Scatterer/Scatterer.dll",
"Shabby": "GameData/Shabby/Plugins/Shabby.dll",
"SmokeScreen": "GameData/SmokeScreen/SmokeScreen.dll",
"SoundtrackEditorForked": "GameData/SoundtrackEditorForked/Plugins/SoundtrackEditorForked.dll",
"SpaceDust": "GameData/SpaceDust/Plugins/SpaceDust.dll",
"ButtonManager": "GameData/SpaceTuxLibrary/Plugins/ButtonManager.dll",
"KSP-ColorPicker": "GameData/SpaceTuxLibrary/Plugins/KSP_ColorPicker.dll",
"KSP-Log": "GameData/SpaceTuxLibrary/Plugins/KSP_Log.dll",
"KSP-PartHighlighter": "GameData/SpaceTuxLibrary/Plugins/KSP_PartHighlighter.dll",
"SpaceTuxUtility": "GameData/SpaceTuxLibrary/Plugins/SpaceTuxUtility.dll",
"VesselModuleSave": "GameData/SpaceTuxLibrary/Plugins/VesselModuleSave.dll",
"SpeedUnitAnnex": "GameData/SpeedUnitAnnex/Plugins/SpeedUnitAnnex.dll",
"HabUtils": "GameData/StationPartsExpansionRedux/Plugins/HabUtils.dll",
"SystemHeat": "GameData/SystemHeat/Plugin/SystemHeat.dll",
"Trajectories": "GameData/Trajectories/Plugin/Trajectories.dll",
"KerbalAlarmClock": "GameData/TriggerTech/KerbalAlarmClock/KerbalAlarmClock.dll",
"TransferWindowPlanner": "GameData/TriggerTech/TransferWindowPlanner/TransferWindowPlanner.dll",
"Scale": "GameData/TweakScale/Plugins/PluginData/Scale.PartDB.19x.dll",
"TweakScaleCompanion-Frameworks": "GameData/TweakScaleCompanion/Frameworks/TweakScaleCompanion_Frameworks.dll",
"TweakScalerWaterfallFX": "GameData/TweakScaleCompanion/Frameworks/Waterfall/TweakScalerWaterfallFX.dll",
"TweakScalerWaterfallFXIntegrator": "GameData/TweakScaleCompanion/Frameworks/Waterfall/PluginData/TweakScalerWaterfallFXIntegrator.dll",
"TweakScaleCompanion-PKMC": "GameData/TweakScaleCompanion/PKMC/TweakScaleCompanion_PKMC.dll",
"UniversalStorage2": "GameData/UniversalStorage2/Plugins/UniversalStorage2.Unity.dll",
"VesselView": "GameData/VesselView/Plugins/VesselView.dll",
"VesselViewRPM": "GameData/VesselView/Plugins/VesselViewRPM.dll",
"VVDiscoDisplay": "GameData/VesselView/Plugins/VVDiscoDisplay.dll",
"VVPartSelector": "GameData/VesselView/Plugins/VVPartSelector.dll",
"Waterfall": "GameData/Waterfall/Plugins/Waterfall.dll",
"WhoAmI": "GameData/WhoAmI/WhoAmI.dll",
"WildBlueTools": "GameData/WildBlueIndustries/000WildBlueTools/Plugins/WildBlueTools.dll",
"KerbalActuators": "GameData/WildBlueIndustries/001KerbalActuators/Plugins/KerbalActuators.dll"
},
"installed_modules": {
"BreakingGround-DLC": {
"install_time": "2023-02-22T21:11:38.4744623-06:00",
"source_module": {
"spec_version": "v1.28",
"identifier": "BreakingGround-DLC",
"name": "BreakingGround-DLC",
"abstract": "An official expansion pack for KSP",
"author": [
"SQUAD"
],
"version": "1.7.1 (unmanaged)",
"license": [
"restricted"
],
"download_content_type": null,
"kind": "dlc"
},
"installed_files": {}
}
},
"installed_files": {},
```

`installed_modules` contains only Breaking Ground, but there are many, many entries in `installed_dlls`. `GrannusExpansionPack` and `GPPSecondary` are in neither list, so they may be getting pulled in as dependencies somehow. What is in the changset when you're getting the message about GEP and GPPSecondary?

**Comment by TheDancingFerret** [comment]
Trying to install literally anything would cause the issue. I did uninstall (via CKAN) then try to reinstall Mechjeb2, which has no dependencies, and it still had the error. 

That being said, I think the issue is resolved(?)

I think I gave you the wrong registry.json. The likely correct one is here: https://drive.google.com/file/d/1qQJZASjExjGGlh_1ffh8o0nD1Fpp6gG2/view?usp=sharing
 
What I think happened is I must have cut instead of copied it when I was going to upload it. I had to step away from the computer midway through that process so when I came back CKAN must have regenerated the file so I ended up posting that one.
 
 When I looked at CKAN after that all of the mods had that "AD" flag, which seems consistent with what you read from the first file.
 
 I copied the one I linked in this post back into the CKAN folder and restarted it, and that removed all of the AD flags. However, now it is no longer throwing the error about GEP and GPP and is installing new mods without issue.
 
The only thing more frustrating than something not working and having no idea why is something working and having no idea why.

Thanks for your attention to this HebaruSan. I think I'm good with all this now, although if you want to look into this more for whatever reason I'll happily provide more info.
